crates/orchestrator/src/network.rs

```diff
@@ -61,7 +62,27 @@ macros::create_add_options!(AddCollatorOptions {
     chain_spec_relay: Option<PathBuf>
 });
 
+impl<T: FileSystem> Drop for Network<T> {
+    fn drop(&mut self) {
+        // Below works only in multi-threaded runtime.
+        // In order to not lock other async threads move this thread to the blocking pool.
+        let result = tokio::task::block_in_place(|| {
+            tokio::runtime::Handle::current().block_on(async { self.shutdown().await })
+        });
+
+        if let Err(err) = result {
+            error!("Error shutting down the network - {:?}", err);
+        };
+    }
+}
+
 impl<T: FileSystem> Network<T> {
+    async fn shutdown(&self) -> Result<(), anyhow::Error> {
```

Comment:
Not sure if this method is really needed.
Maybe I could just call `self.dump_logs()` 

```suggestion
     // Teardown the network
    pub async fn destroy(&self) -> Result<(), ProviderError> {
        self.dump_logs().await?;
        self.ns.destroy().await
    }```